### ISSUE TITLE
Support Faraday 2

### ISSUE THREAD

**Issue by tagliala** [elastic/elastic-transport-ruby#21]
Hi 👋🏼👋🏼

The final version of [Faraday 2.0](https://github.com/lostisland/faraday) has just been released and I'm asking if there are plans to support it

The breaking changes are listed in the [Upgrading Guide](https://github.com/lostisland/faraday/blob/main/UPGRADING.md#faraday-20)

**Comment by picandocodigo** [comment]
Hi @tagliala,
Thanks for the heads up, I'll look into this :+1:

**Comment by jaredbeck** [comment]
All of the Google gems (`googleauth`, `gapic-common`, ..) now support both Faraday 1 and 2. I haven't read through their source, so I don't know if it'll be possible for `elasticsearch-transport` to adapt their technique, and support both versions, but it would be nice, making the upgrade easier for apps.

**Comment by tagliala** [comment]
As far as I can tell from 93f4108f58efa15b9987807235d82e07a77bd3ee, this is going to support both Faraday 1 and 2, but there is an upstream dependency locking the upgrade process

Ref: #27

**Comment by jaredbeck** [comment]
> .. there is an upstream dependency locking the upgrade process

If you mean Typhoeus, it looks like https://github.com/dleavitt/faraday-typhoeus has been released, and is listed on the [official adapter list](https://github.com/lostisland/awesome-faraday#adapters).

> As far as I can tell from https://github.com/elastic/elastic-transport-ruby/commit/93f4108f58efa15b9987807235d82e07a77bd3ee, this is going to support both Faraday 1 and 2,

Sweet.

**Comment by picandocodigo** [comment]
Hi,
We're working on the migration and we'll support Faraday 1 and Faraday 2 once that work is ready.
I will comment on this issue once we have a release with this change.
Thanks!

**Comment by picandocodigo** [comment]
I've published a pre-release of `8.1.0` with support for Faraday 2. I'll still be doing more testing with this version before committing to `8.1.0 GA`. You can see the docs [here](https://github.com/elastic/elastic-transport-ruby/releases/tag/v8.1.0.pre1) and check it out on Ruby gems with `gem install --pre elastic-transport` or add it to your gemfile:
`gem 'elastic-transport', '8.1.0.pre1'`

Please check it out if you have the chance and add a new issue if you encounter any problems. Thanks!

**Comment by picandocodigo** [comment]
Version `8.1.0` [has been released](https://github.com/elastic/elastic-transport-ruby/releases/tag/v8.1.0). I've been testing this with the Elasticsearch and Enterprise Search clients, and everything is looking good.
I'm going to close this, but if you find any problems, please open a new issue on this repository. Thanks!
